os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -275,52 +276,78 @@ static void up_dumpstate(void)
 
 	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
 	ustacksize = (uint32_t)rtcb->adj_stack_size;
+
 	lldbg("sp:     %08x\n", sp);
 
+	/* Show the interrupt number, handler and stack information
+	 * if we are in an interrupt handler (g_irq_nums[1] != 0)
+	 */
+
+	int assertInIrq = g_irq_nums[1];
+
+	if (assertInIrq) {
+#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
+#ifdef CONFIG_STACK_COLORATION
+		if (up_check_nestirqstack()) {
+			lldbg("Code asserted in nested IRQ state!\n");
+		}
+#endif
+#else
+		lldbg("Code asserted in IRQ state!\n");
+#endif
+		lldbg("IRQ num: %d \n", g_irq_nums[1]);
+		lldbg("IRQ handler: %08x \n", g_irqvector[g_irq_nums[1]].handler);
+#ifdef CONFIG_DEBUG_IRQ_INFO
+		lldbg("IRQ name: %s \n", g_irqvector[g_irq_nums[1]].irq_name);
+#endif
+	} else {
+		lldbg("Code asserted in normal thread!\n");
+	}
+#if CONFIG_ARCH_INTERRUPTSTACK > 3
```

Comment:
`CONFIG_ARCH_INTERRUPTSTACK ` must not be checked here. Please revert it to line 304.

Reply:
Agreed, Modified the code